### PR TITLE
Add a PPX for deriving Irmin generics

### PR DESCRIPTION
**Pull request by CraigFe**
This imports [`ppx_irmin`](https://github.com/CraigFe/ppx_irmin), a PPX deriving plugin for generating Irmin generics.

### CODE REVIEW

File: ppx_irmin.opam [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+bug-reports: "https://github.com/CraigFe/ppx_irmin/issues"
+license: "BSD-2"
+dev-repo: "git+https://github.com/CraigFe/ppx_irmin.git"
+
+build: [
+  [ "dune" "build" "-p" name "-j" jobs ]
+]
+
+run-test: [
+  [ "dune" "runtest" "-p" name "-j" jobs ]
```
Comment (samoht) [inline ppx_irmin.opam]: The other packages have [these test runes](https://github.com/mirage/irmin/blob/66bcb96e682d67db837c958a48921c62de015fa4/irmin.opam#L10-L14) instead.

File: src/ppx_irmin/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+You can now use `[@@deriving irmin]` after a type declaration in your code to automatically derive
+an Irmin generic with the same name.
+
+
+## Specifics
+
+`ppx_irmin` supports all of the type combinators exposed in the
+[Irmin.Type](https://docs.mirage.io/irmin/Irmin/Type/index.html) module (basic types, records,
+variants (plain and closed polymorphic), recursive types etc.). Irmin types must fully grounded (no
```
Comment (samoht) [inline src/ppx_irmin/README.md]: I guess *must*  is a current limitation of the implementation, not a general limitation. Maybe be a bit more explicit about it (e.g. *not currently supported* or something like this).

File: src/ppx_irmin/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# ppx_irmin
```
Comment (samoht) [inline src/ppx_irmin/README.md]: can you move this at the toplevel and/or add a link from the main README.

File: src/ppx_irmin/README.md (hunk 2 above)
Comment (CraigFe) [reply src/ppx_irmin/README.md]: Indeed, the [analogous tool in Haskell](http://dreixel.net/research/pdf/gdmh.pdf) does support higher-kinded generics via a `Par` combinator. Clarified in https://github.com/mirage/irmin/pull/919/commits/e77dffbd68874ca09485577755ec8e5de5fc05e7; let me know what you think.

File: src/ppx_irmin/README.md (hunk 3 above)
Comment (CraigFe) [reply src/ppx_irmin/README.md]: Moved to `./README_PPX.md` and added some cross-referencing in https://github.com/mirage/irmin/pull/919/commits/0656db82386bc41c3663e55a491940fa23abfa14.

**Comment by samoht** [comment]
Thanks for the great addition to Irmin!

**Pull request merged**